andela/ah-kgl-wakanda-backend
# #166057229 Protect routes

#### What does this PR do?

- Protects unprotected routes

- Gives only the owner of an item the right to update or delete it

- Tests both functionalities for all routes

- Fixes the ID constancy in the models and tests.

#### Description of Task to be completed?

- Protected all unprotected routes

- Added restrictions for non-owners of the items to delete/update items like article and comments.

- Tested both functionalities in all routes.

- Fixed the ID constancy in the models and the tests.

#### How should this be manually tested?

- Through the existing routes, if the user is not allowed to access, delete or update items, they should not make it access the corresponding routes.

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- N/A

#### Screenshots (if appropriate)

- N/A

#### Questions:

- N/A
